### DISCUSSION TITLE
Getting type 'String' is not a subtype of type 'Map<String, dynamic>' in type cast in Android only

### DISCUSSION THREAD

**Discussion by colin-young** [rrousselGit/riverpod#3621] [Q&A]
I have been developing on MacOS and everything has been working fine. When I run on an Android emulator I'm getting "type 'String' is not a subtype of type 'Map<String, dynamic>' in type cast".

```
Widget build(BuildContext context, WidgetRef ref) {
  final workoutDefinitionsResult = ref.watch(getLastWorkoutDateProvider);
  final lastWorkoutRecordResult = ref.watch(getLastworkoutRecordProvider);
  final isCompleteResult = ref.watch(isWorkoutCompleteProvider(
      workoutRecordId: switch (lastWorkoutRecordResult) {
    AsyncValue(:final value?) => value.id, // error appears to be here
    _ => -1
  }));
  // ...
```

Any suggestion what I might be doing wrong? Or where to begin looking?

**Comment by gdurandrexel** [comment]
Map<String, dynamic> is typically the type for a json object deserialized with jsonDecode. At some point you receive a String that is not deserialized or a json object when you expect a String.
As to why it happens on mac and not android I wouldn't know. Perhaps the remote call fails because of some local network-related reason and the response body you receive is not what you expect.

If your providers provide strongly typed values (not dynamic or Object) the error should appear before in the processing.

**Reply by colin-young** [reply]
I'm fetching data using Sembast from a local file, and I believe all my providers are using strongly typed values, although I am going to check that again.

The frustrating bit is I have not, so far, been able to figure out what the value is that is causing the problem, or where it is coming from. To be clear, I am only seeing this error when attempting to run the app on the Android emulator. I haven't tried on a real device yet. The MacOS app works fine.

**Reply by colin-young** [reply]
Further update: I do not get the error on a real device. Weird.
